src/Expr.cc

```diff
@@ -4254,6 +4252,19 @@ Val* RecordCoerceExpr::Fold(Val* v) const
 					rhs = new_val;
 					}
 				}
+			else if ( BothArithmetic(rhs_type->Tag(), field_type->Tag()) &&
+				      ! same_type(rhs_type, field_type) )
+				{
+				if ( Val *new_val = check_and_promote(rhs, field_type, false, op->GetLocationInfo()) )
+					{
+					// Don't call unref here on rhs because check_and_promote already called it.
+					rhs = new_val;
+					}
+				else
+					{
+				    break;
```

Comment:
There was a pattern I was following from elsewhere, but now I can't remember where it was I got it from. Is it reasonable to have it fail out during runtime (i.e. not during initialization), since it would have been an error anyways?

Reply:
The only issue with calling `RuntimeError` is that it ends the parsing immediately on any error, so the second intentional error in the test script doesn't ever get hit.

Reply:
> Is it reasonable to have it fail out during runtime (i.e. not during initialization), since it would have been an error anyways?

Usually best to fail out early and only during parse-time for things we can catch at that point, and then at run-time only if something seems really irrecoverable and we couldn't have caught it at parse-time.

> The only issue with calling `RuntimeError` is that it ends the parsing immediately on any error, so the second intentional error in the test script doesn't ever get hit.

It shouldn't end the *parsing*, but it does return early from evaluating event handlers at run-time, so you can change the test script to use two (or more) `zeek_init()` event handlers.